src/inject.cc

```diff
@@ -342,11 +342,16 @@ bool read_header()
 {
   mystring cur_line;
   mystring whole;
+  bool first = true;
   for (;;) {
     if (!fin.getline(cur_line))
       cur_line = "";
     if(!cur_line || cur_line == "\r")
       break;
+    if(first && (cur_line.find_first_of("From ", 0) == 0 ||
+       cur_line.find_first_of(">From ", 0) == 0))
```

Comment:
Actually, on further review, this doesn't do what you want it to. This will trigger if the first character of the first line is either F, r, o, m, >, or " " (find_first_of treats the first argument as a set of characters, not a string). This really needs the starts_with method to be useful.